apps/trackers/product_definition_handlers/fedramp_handler.py

```diff
@@ -0,0 +1,84 @@
+        #     - base (seems to run all handlers)
+        #     - ubi_handler
+        #     - unacked_handler
+        #   - TrackerFileSuggestionView.post (seems to do the beginning of sfm2's TrackerFileSuggestionView)
+        #   - FEDRAMP_OVERRIDES probably needs a new fedramp_handler
+        #   - The logic around unacked seems to be different - need deeper understanding to understand differences and similarities.
+
+    def compliance_priority_fedramp_streams(ps_module: PsModule, affect: Affect) -> list[str]:
+        # TODO:
+        #  - if ps module in compliance priority
+        #    - if
+        #      - list of components is empty
+        #      - OR ps component in compliance_priority.yml's component for that module
+        #  - return sorted streams for compliance_priority.yml's ps module
+        #  -
+        # TODO ^ doesn't that just mean if is_compliance_priority, return related compliance priority streams?
+        #  -> Probably yes!
+        if affect.is_compliance_priority:
+            # TODO returns list[str] but has to return list[proprer stream object] for natural sorting to work
+            return CompliancePriority.objects.get(ps_module=ps_module).streams
+
+    def get_offer(self, affect: Affect, impact: Impact, ps_module: PsModule, offers):
+        # compliance priority / fedramp special handling
+        #
+        # Only apply this special handling when fedramp stream is
+        # configured in the compliance_priority list.
+        if FedrampHandler.compliance_priority_fedramp_streams(ps_module, affect) and impact in self.FEDRAMP_OVERRIDES:
+            # TODO: SFM2 Overwrites whatever changes are made by the ubi_handler.py-equivalent code block.
+            #       This is fundamentally incompatible with the current code structure in OSIDB.
+            #       Also, it makes any policy even more unclear.
+            #       TODO: Either change OSIDB's structure to have one common handler for both UBI + Fedramp
+            #             OR somehow check for fedramp within ubi_handler (bleh!)
+            #             OR decide that OSIDB will do it differently than SFM2 (as it is now!)
+            #             ^ TODO/NOTE: I prefer the last option, but don't understand implications.
+            #       TODO/NOTE: Also, I *think* that it doesn't matter much because all streams
+            #                  higher than selection are then selected anyway, so it seems the
+            #                  intent is to select whole ranges starting with the oldest in the
+            #                  respective policy files (ubi_packages.yml and compliance_priority.yml).
+
+            # TODO: Why is this in SFM2? Why can't we just preselect all, given the fact that
+            #       the block after that tries to select anything newer than preselected anyway?!
+            # If multiple fedramp streams are configured, only seed the
+            # streams_to_preselect with the first oldest one.
```

Comment:
@osoukup A bunch of questions re. stream preselection.

First intro: In SFM2 all stream preselection is done in a single function. In OSIDB, the functions are mutually independent, but contributing to a shared dict of selected streams. In SFM2, fedramp handling cancels whatever ubi handling did (overwrites the streams_to_preselect list). In OSIDB, this would be non-trivial.

Questions:

1. Is it a problem? It seems to me that the ubi and fedramp policies just identify some specific streams and then that and all newer streams are selected anyway. Trying to not handle UBI if handling fedramp for a given affect(?) would probably be difficult but probably not too different? (I don't understand the consequences though.)
2. Can we depart in OSIDB from SFM2 in that UBI and Fedramp are independent? That would force the policies behind UBI handling and fedramp handling to be clear. If yes, it will allow us to keep the current structure of OSIDB's product_definition_handlers.
3. Why does SFM2 contain __If multiple fedramp streams are configured, only seed the streams_to_preselect with the first oldest one._? That seems superfluous and prone to a mistake by omission because the following code selects all newer streams anyway. I don't understand the implications/consequences though.

Reply:
Re. 3.: I think it might be because of code complexity and computational complexity. Best solution probably: Update compliance_priority.yml to explicity state the first stream must be the oldest one.

Reply:
:+1: 

I now have tests robust enough and enough understanding (thanks for your 1:1 explanations!) to try to replicate SFM2's algorithm. I'm just starting to try that.

Reply:
It seems I found a sane way to replicate SFM2's algorithm and give FedrampHandler the ability to effectively undo what UnackedHandler or UbiHandler does.

`offer = []` is not very feasible because it is pre-filled before any handler runs, but I also missed some understanding of the algorithm, which I tried to rectify for posterity by creating more educative tests.